src/wallet/wallet.cpp

```diff
@@ -1040,6 +1044,38 @@ CAmount CWallet::GetChange(const CTransaction& tx) const
     return nChange;
 }
 
+
+
+// MVHF-BU Auto wallet backup
+bool CWallet::BackupWalletAuto(const std::string& strDest, int BackupBlock)
+{
+
+    boost::filesystem::path pathBackupWallet = strDest;
+
+    // Test for directory only, if so add default filename
+	if (boost::filesystem::is_directory(strDest))
```

Comment:
Updated BackupWalletAuto function to handle the 4 variants listed in the test design steps.
The testing was updated to test the 4 variants.